infra/modules/gcp-host/main.tf

```diff
@@ -160,7 +143,7 @@ module "api_connector" {
 
   secret_bindings = merge(
     # bc some of these are later filled directly, bind to 'latest'
-    { for k, v in module.secrets[each.key].secret_bindings : k => merge(v, { version_number : "latest" }) },
+   {for k, secrets in local.secrets_to_provision[each.key] : k => merge({ secret_id : module.secrets[each.key].secret_ids_within_project[k], version_number : "latest" }) if (try(!secrets.lockable, true) && try(!secrets.writable, true))},
```

Comment:
I think the `if` could be
`if !contains(keys(local.writable_secrets), secret_id)`, more succinct